JASP-Desktop/widgets/filemenu/filemenu.cpp

```diff
@@ -411,39 +411,33 @@ bool FileMenu::checkSyncFileExists(const QString &path, bool waitForExistence)
 	if (path.startsWith("http"))
 		return true;
 
-	bool exist = false;
-	int maxExistTrials = waitForExistence ? 5 : 1;
-	int existTrials = 1;
+	auto checkExists = [](const QString& path) { return QFileInfo::exists(path); };
+	auto checkNotEmpty = [](const QString& path) { return Utils::getFileSize(path.toStdString()) > 0; };
 
-	while (!QFileInfo::exists(path) && existTrials < maxExistTrials)
+	auto checkFn = [&](bool (*f)(const QString&))
 	{
-		// sometimes the file does not exist temporarily (when using Excel especially)
-		Log::log() << "Could not find sync file" << std::endl;
-		existTrials++;
-		Utils::sleep(100);
-	}
-
-	if (QFileInfo::exists(path))
-	{
-		int trials = 0;
-		while (Utils::getFileSize(path.toStdString()) == 0 && trials < 5)
+		bool check = f(path);
+		if (!check && waitForExistence)
```

Comment:
Checking for `!check` is superfluous right? 
The condition is always checked in the for loop before executing the body: https://en.cppreference.com/w/cpp/language/for
And you condition already includes `!check`.

I would write it as:
```C++
if(!waitForExistence) return f(path);

for (int i = 0; i < 10; i++)
{
	if(f(path))
       return true;
   	Utils::sleep(100);
}
return false;
```

Reply:
I don't really like that programming pattern where stuff is put in booleans or whatever and updated in random loops and you have to read an entire function and understand what is going on. This function is short but you really don't need `bool check` at all.

I think this is much cleaner and easier to understand.

Reply:
(Cleaner except for the messed up tabs but that is because Im typing here in github comment field...

Reply:
During my study I was told to use only one return at the end of a function. But I see your point. Maybe I'm a bit old-fashioned.
I was happy with my lambda function as parameter of a lambda function! Apparently it does not work if the parameter lambda function captures external variables.

Reply:
Well, that makes a lot of sense in C, in C++ less.
And yeah now that you mention that, I have suggested a small extra change to that using the `std::function`. Maybe you can try that?

And what exactly doesn't work with the lambda as a parameter?

Reply:
Compile error. But maybe std::function works beter indeed

Reply:
So does this code compile or not? Because it looks nice and clear!